lib/fxpay/jwt.js

```diff
@@ -20,18 +26,22 @@
     try {
       jwtString = atob(jwtData);
     } catch (error) {
-      settings.log.error('atob() error:', error.toString(),
-                         'when decoding JWT', jwtData);
-      return callback('INVALID_JWT_DATA');
+      newError = new errors.InvalidJWTData(
+                      'atob() error: ' + error.toString() +
+                      ' when decoding JWT ' + jwtData);
+      settings.log.error(newError);
+      return callback(newError);
     }
     var data;
 
     try {
       data = JSON.parse(jwtString);
     } catch (error) {
-      settings.log.error('JSON.parse() error:', error.toString(),
-                         'when parsing', jwtString);
-      return callback('INVALID_JWT_DATA');
+      newError = new errors.InvalidJWTData(
+                        'JSON.parse() error: ' + error.toString() +
+                        ' when parsing ' + jwtString);
+      settings.log.error(newError);
```

Comment:
the only reason the object is assigned (in most cases) is for the logging. Hmm, I guess I could log in the constructor but that might be weird since constructing an error doesn't always mean you are returning an error.